xdsl/dialects/dlti.py

```diff
@@ -30,6 +37,23 @@ class DataLayoutEntryAttr(ParametrizedAttribute):
     key: Attribute
     value: Attribute
 
+    def __init__(
+        self,
+        key: StringAttr | TypeAttribute | str,
+        value: Attribute | str | int | float,
+    ):
+        if isinstance(key, str):
+            key = StringAttr(key)
+
+        if isinstance(value, str):
+            value = StringAttr(value)
+        elif isinstance(value, int):
+            value = IntegerAttr(value, i32)
+        elif isinstance(value, float):
+            value = FloatAttr(value, Float32Type())
+
+        super().__init__(key, value)
+
     def verify(self) -> None:
         if not isinstance(self.key, StringAttr | TypeAttribute):
```

Comment:
feels like this should also be a constraint on the key directly?

Reply:
I think this is what we discussed in terms of needing Attribute on the key (for now)?

Reply:
Yep